pkg/cnb/create_builder.go

```diff
@@ -30,14 +33,29 @@ type RemoteBuilderCreator struct {
 	ImageSigner       cosign.BuilderSigner
 }
 
-func (r *RemoteBuilderCreator) CreateBuilder(ctx context.Context, builderKeychain authn.Keychain, stackKeychain authn.Keychain, fetcher RemoteBuildpackFetcher, clusterStack *buildapi.ClusterStack, spec buildapi.BuilderSpec, serviceAccountSecrets []*corev1.Secret) (buildapi.BuilderRecord, error) {
-	buildImage, _, err := r.RegistryClient.Fetch(stackKeychain, clusterStack.Status.BuildImage.LatestImage)
+func (r *RemoteBuilderCreator) CreateBuilder(
+	ctx context.Context,
+	builderKeychain authn.Keychain,
+	stackKeychain authn.Keychain,
+	fetcher RemoteBuildpackFetcher,
+	clusterStack *buildapi.ClusterStack,
+	spec buildapi.BuilderSpec,
+	serviceAccountSecrets []*corev1.Secret,
+	resolvedBuilderTag string,
+) (buildapi.BuilderRecord, error) {
 
+	buildImage, _, err := r.RegistryClient.Fetch(stackKeychain, clusterStack.Status.BuildImage.LatestImage)
+	if err != nil {
+		return buildapi.BuilderRecord{}, err
+	}
+	runImage, _, err := r.RegistryClient.Fetch(stackKeychain, clusterStack.Status.RunImage.LatestImage)
```

Comment:
Would it be possible to only save the image if it is in a different registry than the builder?

Reply:
I was thinking you might still run into a case where the pull rules could be different for a different repo

Reply:
Fair. There is a trade-off from having the run-image be relocated everywhere, though.

Reply:
I think I'm ok with just checking the registry and saying you need to be able to pull both. Maybe putting the original image in the run image mirror would also help as a backup

Reply:
I was thinking the downside is more about more confusing provenance. It is useful to see that the run image is `my-registry.com/my-stack/specific-stack-name/run` vs `my-registry.com/some-team/custom-builder/run`.

but, I agree it is not a fatal flaw.

Reply:
ya that is definitely a concern, but I was hoping the tags on the relocated run image would simplify this. It also makes registry management a bit easier so you can easily see the run image associated with a particular builder